### DISCUSSION TITLE
Can't transmit the same number more than once?

### DISCUSSION THREAD

**Discussion by SteveMyers75** [bengtmartensson/IrScrutinizer#528] [Hardware issues] [answered]
I'm using IrScrutinizer v2.4.1 with GirsLite 1.0.5 on an Arduino UNO and ran into an issue while testing output.
When I try to transmit the same number twice, it only sends the 1st one.

Example: 
I'm setting the time on an IR Clock and try to set it to **08:33**, but it gets stuck at **08:3** and it doesn't receive the 2nd "**3**"
I can tell it to transmit **3** over and over, but the clock never "sees" it. 
If I tell it to Transmit any other number that is not **3**, it sends it, but of course the time is now wrong.

I tried using the **Scrutinize Signal Transmit** button and the **Scrutinize Remote** Right-Click **Transmit Selected** with no luck.  

This happens with any number I try to send more than one time in a row, not just the **3**
I can send 12:12, 03:03, 06:56, 05:15... but not 0**4:4**5 **11**:20  12:**33**   

When I use the IR Keyboard I'm trying to emulate, I can set the time to **08:33** with no problem.
I've tried re-installing the software and re-booting the PC but no matter what, I can't seem to send the same Number two times.

Any ideas of what am I doing wrong  or have setup wrong?

The device I'm emulating is a non-standard protocol device and is not reconized.

For example,  
#5 =  0000 006A 0000 0007 0060 001F 001F 001F 003C 001F 001F 001F 0060 001F 003C 001F 003C 09DD
#6 = 0000 006A 0000 0006 0060 001F 001F 001F 003C 001F 001F 001F 0060 009A 003C 09D9
#7= 0000 006A 0000 0007 0063 001F 001F 001F 003B 001F 001F 001F 001F 001F 001F 0099 001F 0A0A

**Comment by bengtmartensson** [comment]
If I, or  anyone else, should have a chance to help you, you have to give more details, in particular the exact signals (protocol and parameters) used.

> When I try to transmit the same number twice, it only sends the 1st one.

Sounds like the receiving device expects a toggle of some sort, but you are not sending it correctly.

**Comment by bengtmartensson** [comment]
Try to capture a key, for example 3, multiple times. Then we can see if there is a toggle,

**Reply by SteveMyers75** [reply]
3-#1
0000 006D 001F 0000 0000 0216 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 0141 006A 001E 001E 001E 001E 006A 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 006A 0141 09BA

3-#2
0000 006D 001F 0000 0000 0217 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 0217 006A 001E 001E 001E 001E 006A 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 006A 010C 09BA

3-#3
0000 006A 001E 0000 006D 001F 001F 001F 0038 001F 001F 001F 001F 001F 001F 001F 001F 001F 006D 001F 0000 0A0A 006D 001F 001F 001F 0038 001F 001F 001F 001F 001F 001F 001F 001F 001F 006D 001F 0000 02C1 006D 001F 001F 001F 001F 006D 001F 001F 001F 001F 001F 001F 006D 001F 0000 0A0A 006D 001F 001F 001F 0038 006D 0115 0A0A

3-#4
0000 006D 001F 0000 0000 0220 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 03E2 006A 001E 001E 001E 001E 006A 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 006A 010C 09BA

3-#5
0000 006D 001F 0000 0000 0220 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 03E2 006A 001E 001E 001E 001E 006A 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 006A 010C 09BA

**Comment by SteveMyers75** [comment]
It must be something wrong with my Arduino.

If I: 
  1. Capture a number  
  2. Send it 
  3. Capture the same number again 
  4. Send it 
It works.

If I capture the same number many times in a row and try to send all of them back , it only receives the 1st one.

Either way, it's not a IrScrutinizer issue - it's my setup.
I'll have to work on this again tomorrow....it's 4am here now.

**Comment by bengtmartensson** [comment]
>    Capture a number
>    Send it
 >   Capture the same number again
 >   Send it
>    It works.
>
> If I capture the same number many times in a row and try to send all of them back , it only receives the 1st one.

If the device uses toggles, this is completely plausible.

> #3-1
> 0000 006D 001F 0000 0000 0216 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 001E 001E 001E 001E 001E 001E 001E 001E 001E 006A 001E 0000 0141 006A 001E 001E 001E 001E 006A 001E 001E 001E 001E 001E 001E 006A 001E 0000 09BA 006A 001E 001E 001E 0036 006A 0141 09BA

These signals are nonsensical; there can be 0000 only on position 0, 3 and 4.

Please do the following: Capture all signals, every key twice in succession. Post it. Preferably as gist, not cut-n-paste. Then we can probably identify the protocol.

**Comment by SteveMyers75** [comment, marked as answer]
**Thanks** to your help, I was able to get the entire keyboard to decode.
It works fine when I transfer it to .IRPlus format and load it into my Android Phone.

After you said that only bytes 00,03,04 should be 0000, I re-loaded each key into Scrutinize Signal and let it fix the 0000 errors it found, Transmitted them and they worked fine.

I still have no idea as to what Protocol these keyboards use, but that's something I may never figure out.,....**but it works!**

Here is the complete usable SWK-8630 keyboard. 
https://gist.github.com/SteveMyers75/3be7bd67f5f7e537782aa86f66800169
